iwyu.cc

```diff
@@ -3393,14 +3417,25 @@ class IwyuAstConsumer
     ParseFunctionTemplates(context.getTranslationUnitDecl());
 
     TraverseDecl(context.getTranslationUnitDecl());
+   
+    const set<const FileEntry*>* const files_to_report_iwyu_violations_for
+          = preprocessor_info().files_to_report_iwyu_violations_for();
+
+    // Some analysis, such as UsingDecl resolution, is deferred until the
+    // entire AST is visited because it's only at that point that we know if
+    // the symbol was actually used or not.
+    // We perform that analysis here before CalculateAndReportIwyuViolations.
+    for (Each<const FileEntry*> file(files_to_report_iwyu_violations_for);
```

Comment:
Range-for again. We're slowly trying to transition to range-for as we change code, but we don't want to do a global conversion just yet.